python/ray/serve/_private/controller.py

```diff
@@ -477,21 +483,39 @@ def _create_control_loop_metrics(self):
         )
 
     def _recover_config_from_checkpoint(self):
-        checkpoint = self.kv_store.get(CONFIG_CHECKPOINT_KEY)
-        if checkpoint is not None:
+        serve_config, deployment_time = self._get_curr_serve_config()
+        if serve_config is not None:
             logger.info(
-                "Recovering config from checkpoint.", extra={"log_to_stderr": False}
+                "Recovered config from checkpoint.", extra={"log_to_stderr": False}
             )
+            self.deploy_config(serve_config, deployment_time=deployment_time)
+
+    def _get_curr_serve_config(self) -> Tuple[Optional[ServeDeploySchema], float]:
+        """Gets the current active Serve config.
+
+        Returns: Tuple containing:
+            1. A Serve config, if the GCS contains a config checkpoint. None
+               otherwise
+            2. A deployment timestamp, if the GCS contains a config checkpoint.
+               0 otherwise.
+        """
+
+        checkpoint = self.kv_store.get(CONFIG_CHECKPOINT_KEY)
+        if checkpoint is not None:
+
             deployment_time, target_capacity, config_checkpoints_dict = pickle.loads(
                 checkpoint
             )
-            self.deploy_config(
+
+            return (
                 ServeDeploySchema(
                     applications=list(config_checkpoints_dict.values()),
                     target_capacity=target_capacity,
                 ),
-                deployment_time=deployment_time,
+                deployment_time,
             )
+        else:
+            return (None, 0)
```

Comment:
Nitpick (non-blocker): Weird the linter didn't caught this. But should probably use `0.0` instead of `0` given it should return a float 🙃

Reply:
Thanks for the catch, I changed it.